### ISSUE TITLE
Intellij 2018.1 JBehave Syntax Support locks up on autocomplete

### ISSUE THREAD

**Issue by ron-noble** [stefanschenk/IntelliJBehave#12]
Hi, 

Currently using your plugin to support developing features in JBehave. The plugin works but when writing an ExamplesTable it causes the autocomplete to pop up and this will lock up Intellij. 

The step that can reproduce this issue every time

@Then("the $messageType is verified against the transaction log with values: $values")

When writing the following Syntax, values can be a String or an ExamplesTable, messageType is a String

Then the MSG1 is verified against the transaction log with values:
    | KEY  | VALUE |
    | MtDt.Status | REQUESTED

If I type REQUESTED it seems to pop up with suggestions for autocomplete for all my steps as the matchers are matching everything because it's including whitespace.

![image](https://user-images.githubusercontent.com/11923249/40726851-80ef1c98-641e-11e8-9839-8034effc69ad.png)

You can see it matches on the step itself but it tries to treat "MSG1 is verified against the transaction log with values:
    | KEY  | VALUE |
    | MtDt.Status | REQUESTED"

As the first parameter in the step $messageType

Taking a sample of CPU time in JVisualVM shows the following to processes switching constantly

com.intellij.psi.codeStyle.NameUtil.nextWord() 
BaseDataReader.doRun()

I have added a Thread dump here 

https://pastebin.com/tezaC2nt

P.S 

Updated to 2018.1.4 
Autocomplete actually shows the wrong suggestions, when clicking through the autocomplete text does not match the suggestion shown in the pop up box.

The problem is with typing characters after the auto complete box is shown. The plugin will not lock up if you type one character at a time and close the autocomplete dialog after each character is typed.

**Comment by ron-noble** [comment]
Seems like something between calling multiResolve and processStepDefinition is hanging but I'm yet to work out what.

[START] StepPsiReference.multiResolve() 1527776178262
[END] StepPsiReference.multiResolve()  1527776178262
//Hangs here 
[START] StoryCompletionContributor.processStepDefinition() 1527776364726
[END] StoryCompletionContributor.processStepDefinition() 1527776364726

**Comment by stefanschenk** [comment]
Thanks for reporting this issue. I have some more things to check with intellij 2018. So I will also take a look at this.

**Comment by ron-noble** [comment]
It seems that having 0 whitespace before my example table prevents the suggestions from popping up. 

It will not treat the table as a string which stops it autocompling the table as $messageType. This is the current work around. Seems that the lock up doesn't occur on simpler steps without ExampleTables

![nosuggestions](https://user-images.githubusercontent.com/11923249/40965755-877ac152-68a6-11e8-9038-2551d8185dce.png)

![suggestions](https://user-images.githubusercontent.com/11923249/40965760-89520ada-68a6-11e8-9910-8b1f4925962d.png)

If you have any questions feel free to ask and I'll help if I can

**Comment by cristiadu** [comment]
Here that happens on all cases, not only for example tables. If I'm typing a step into a .story file the autocomplete shows up and freezes intelij. It happens about 8 out of 10 tries =/

I need to keep doing copy and paste of steps if I don't want that to happen

**Comment by ron-noble** [comment]
I believe it'll be to do with the regex that is used in JBehave, it's super wide (.*) because we were executing the wrong steps at times until we changed the StepFinder to use StepFinder.ByLevenshteinDistance

**Comment by cristiadu** [comment]
Yes, It relates to that. All steps that I have which end on a regex expression freezes intelij.

I could just change my steps to not end with a regex, but we created a "retry" step in here which allows us to execute any steps multiple times until it passes or the number of times ends, the matching expression is "Then we retry $times times with an interval of $ms ms between each try: $step". It doesn't make much sense adding something after me passing the steps.

Can't we just keep doing what was happening for Intelij 2017 which doesn't freeze?
